platform/x11/os_x11.cpp

```diff
@@ -113,7 +116,21 @@ void OS_X11::initialize(const VideoMode &p_desired, int p_video_driver, int p_au
 	/** XLIB INITIALIZATION **/
 	x11_display = XOpenDisplay(NULL);
 
-	char *modifiers = XSetLocaleModifiers("@im=none");
+	Bool xkb_dar = False;
+	if (x11_display) {
+		XAutoRepeatOn(x11_display);
+		xkb_dar = XkbSetDetectableAutoRepeat(x11_display, True, NULL);
+	}
+
+	char *modifiers = NULL;
+	if (xkb_dar == True) {
+		modifiers = XSetLocaleModifiers("");
+	}
+	if (modifiers == NULL) {
+		// To support echo events.
+		WARN_PRINT("IME is disabled");
```

Comment:
I wouldn't display a warning here, it will confuse the many users who don't even know what an IME is :)
Maybe add to the verbose output instead:
```
if (is_stdout_verbose()) {
	print_line("IME is disabled");
}